### PR TITLE
DOC: stats: describe loc and scale parameters in the docstring example

### PR DESCRIPTION
**Pull request by ev-br**
so that each distribution has a magic formula `y = (x - loc) / scale` in its docstring.
As suggested in gh-4538.

[ci skip]

### CODE REVIEW

**Comment by pv** [comment]
How about including the definition next to the the Latex formulas for
the PDF/PMF? I think these are what people will first look (because tl;dr).

**Comment by pv** [comment]
For example here http://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.lognorm.html
it should say
```
The probability density function for lognorm is::

    lognorm.pdf(y, s) = 1 / (s*y*sqrt(2*pi)) * exp(-1/2*(log(y)/s)**2)
    y = (x - loc) / scale

for y > 0, s > 0.
```
And similarly repeated in all docstrings --- what `loc` and `scale` mean is otherwise very unclear as the docstring is not self-contained in that case, and explaining the concept in words in the Parameters section is confusing.

**Comment by DanielSank** [comment]
Actually, I think even putting "loc" and "scale" in the TeX strings is still really confusing. Here's the docstring for `numpy.random.normal`:

```python
"""
normal(loc=0.0, scale=1.0, size=None)
<snip>
Parameters
----------
loc : float
    Mean ("centre") of the distribution.
scale : float
    Standard deviation (spread or "width") of the distribution.
<snip>
Notes
-----
The probability density for the Gaussian distribution is

.. math:: p(x) = \frac{1}{\sqrt{ 2 \pi \sigma^2 }}
                 e^{ - \frac{ (x - \mu)^2 } {2 \sigma^2} },

where :math:`\mu` is the mean and :math:`\sigma` the standard deviation.
The square of the standard deviation, :math:`\sigma^2`, is called the
variance.
"""
```

The args are called "loc" and "scale". Then in the parameters section they are renamed "mean" and "standard deviation". Then in the formula they are renamed "mu" and "sigma". This would be a lot more clear if the docstring read as follows:

```python
"""
normal(mu=0.0, sigma=1.0, size=None)
<snip>
Parameters
----------
mu : float
    Mean ("centre") of the distribution.
sigma : float
    Standard deviation (spread or "width") of the distribution.
<snip>
Notes
-----
The probability density for the Gaussian distribution is

.. math:: p(x) = \frac{1}{\sqrt{ 2 \pi \sigma^2 }}
                 e^{ - \frac{ (x - \mu)^2 } {2 \sigma^2} },

where :math:`\mu` is the mean and :math:`\sigma` the standard deviation.
The square of the standard deviation, :math:`\sigma^2`, is called the
variance.
"""
```
In short, use the *same* names *everywhere*.

**Comment by pv** [comment]
You're right that the notation should be made consistent. The arguments need to be called `loc` and `scale` consistently (they are keyword arguments); relationship to the mean and std can be explained in the text.

**Comment by DanielSank** [comment]
@pv I see that the names can't be changed easily (1). The easiest thing to do here would be to just rewrite the docstring to use "loc" and "scale" everywhere, instead of renaming them "mean" and "standard deviation" in the Parameters section and then "mu" and "sigma" in the formula.

> relationship to the mean and std can be explained in the text

That's precisely what I don't like about the current version. They are explained in the docstring, but the names being inconsistent puts an unnecessary cognitive load on the reader.

(1) I'm not sure what you mean by "they are keyword arguments, so cannot be renamed". *Every* argument in a python function call can be called in any order if called as keyword arguments. I forgot this last night when I made my post.

**Comment by ev-br** [comment]
OK, bubbled up the explanation to the Notes section, closer to the formulas for pdf/pmf so that it's not hidden in the wall-of-text of an example. 

A potentially contentious point is that I also standartized the formulas for `gamma` and `expon`, trading the one-off scale descriptions for a generic one.

Also included a nano-cleanup of removing a now-unused key from the docstring dictionary.

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -231,12 +207,19 @@ def instancemethod(func, obj, cls):
 >>> ax.hist(r, normed=True, histtype='stepfilled', alpha=0.2)
 >>> ax.legend(loc='best', frameon=False)
 >>> plt.show()
+
+"""
+
+_doc_default_locscale = """\  
+The probability density above is defined in the "standartized" form. To shift
```
Comment (rgommers) [inline scipy/stats/_distn_infrastructure.py]: Typo, should be "standardized".

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -344,7 +326,17 @@ def instancemethod(func, obj, cls):
 
 >>> r = %(name)s.rvs(%(shapes)s, size=1000)
 """
+
+
+_doc_default_discrete_locscale = """\
+The probability mass function above is defined in the "standartized" form.
```
Comment (rgommers) [inline scipy/stats/_distn_infrastructure.py]: Smae here, "standardized".

**Comment by rgommers** [comment]
Location of the sentence explaining it looks good to me.

**Comment by rgommers** [comment]
The changes to ``gamma`` and ``expon`` look good to me. No need to treat ``scale`` differently there. 

Good to merge I'd think.

**Comment by pv** [comment]
I still think it could be slightly better to change the formulas rather than add some new text. For example,
```
    norm.pdf(x) = exp(-y**2/2)/sqrt(2*pi*scale)
    y = (x - loc) / scale
```
and
```
    alpha.pdf(x, a) = 1/(y**2*Phi(a)*sqrt(2*pi)) * exp(-1/2 * (a-1/y)**2) / scale
    y = (x - loc) / scale
```
```
    cauchy.pdf(x) = 1 / (pi * (1 + y**2)) / scale
    y = (x - loc) / scale
```
This way, there's less to read, and in cases where scale and loc have standard meanings, they are obvious for people looking at the formulas.

**Comment by ev-br** [comment]
Just to put this in context: the difference is between what Pauli writes above and: 

      alpha.pdf(x, a) = 1/(x**2*Phi(a)*sqrt(2*pi)) * exp(-1/2 * (a-1/x)**2)

The probability density above is defined in the "standardized" form. To shift and/or scale the distribution use the ``loc`` and ``scale`` parameters. Specifically, ``alpha.pdf(x, a, loc, scale)`` is identically equivalent to ``alpha.pdf(y, a) / scale`` with ``y = (x - loc) / scale``.

**Comment by ev-br** [comment]
FWIW, I personally have a slight preference against making formulas any busier. For those who are not immediately familiar with loc-scale family, adding extra symbols, `y` and `scales` might be a trifle more confusing. But this is no more than a slight preference for a specific shade for the bikeshed's color anyway.

File: scipy/stats/_distn_infrastructure.py (hunk 2 above)
Comment (ev-br) [inline scipy/stats/_distn_infrastructure.py]: Good catch, fixed!

File: scipy/stats/_continuous_distns.py [hunk 3]
```diff
@@ -947,13 +987,11 @@ class expon_gen(rv_continuous):
...
     The probability density function for `expon` is::
 
-        expon.pdf(x) = lambda * exp(- lambda*x)
+        expon.pdf(x) = exp(-x)
 
     for ``x >= 0``.
 
-    The scale parameter is equal to ``scale = 1.0 / lambda``.
-
-    `expon` does not have shape parameters.
```
Comment (josef-pkt) [inline scipy/stats/_continuous_distns.py]: Does this still mention that the rate parameter  `lambda =  1 / scale` for the common parameterization, see #4538

File: scipy/stats/_continuous_distns.py [hunk 4]
```diff
@@ -3717,6 +3850,8 @@ class rice_gen(rv_continuous):
 
     `rice` takes ``b`` as a shape parameter.
 
+    %(after_notes)s
+
```
Comment (efiring) [inline scipy/stats/_continuous_distns.py]: If rice is left as-is, then I would add something like this:
```
If you use the ``scale`` parameter, you will normally need to substitute ``b/scale`` for the ``b`` shape parameter.
``loc`` should normally be left at its default value, 0.
```

Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Honestly, I don't think we can change rice itself. Backwards compat is one reason, a deeper one is I'm not sure it's too deep of a rabbit hole to start special-casing the behavior of individual distributions with respect to shifting/scaling. (This equally applies to `expon` @josef-pkt mentiones above in this thread: technically it's easy to rename scale to lambda, but FWIW I don't think we should do it). 

So overall I'd view this as a documentation issue. Spefically, how do you see this comment in light of https://github.com/scipy/scipy/pull/4611 -- do you think we should have both @efiring ?

File: scipy/stats/_continuous_distns.py (hunk 3 above)
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: I've added a note:  https://github.com/ev-br/scipy/commit/a8ae65efd736f02ba920496cee97f8f222e9919f

**Comment by ev-br** [comment]
Two more commits: one adds a note about an alternative parameterization to `expon`; the other one adds the loc-scale blurb to `exponnorm` which was merged recently.

**Comment by rgommers** [comment]
needs a rebase

**Comment by ev-br** [comment]
rebased

**Comment by rgommers** [comment]
I also have a slight preference for the form in this PR as is - less busy formula for those who know what loc/scale is, and more context for those who don't. Additionally, editing all the formulas to do ``x -->y`` and ``y = (x - loc) / scale`` is work. So merging this.

**Comment by rgommers** [comment]
Thanks @ev-br, all.

**Pull request merged**